sharktank/sharktank/evaluate/perplexity_vmfb.py

```diff
@@ -0,0 +1,449 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import sys
+import logging
+import json
+import time
+import random
+from datetime import timedelta
+from tqdm import tqdm
+
+import numpy as np
+
+from datasets import load_dataset
+
+import torch
+from torch.nn import CrossEntropyLoss
+
+from sharktank.models.llama.llama import *
+from sharktank.models.mixtral.mixtral import *
+from sharktank.models.grok.grok import *
+
+from ..models.llama.sharding import shard_theta
+
+from sharktank.layers import *
+from sharktank.types import *
+
+from sharktank.utils import cli
+from sharktank.utils.vmfb_runner import *
+from sharktank.utils.load_llm import *
+from sharktank.utils.create_cache import *
+from sharktank.utils.export_artifacts import *
+
+log_levels = {
+    "info": logging.INFO,
+    "debug": logging.DEBUG,
+}
+logger = logging.getLogger("eval")
+
+logger.setLevel(log_levels["info"])
+
+logger.root.handlers[0].setFormatter(
+    logging.Formatter(fmt="\n%(levelname)s:%(name)-8s %(message)s")
+)
+
+__all__ = ["Perplexity", "run_perplexity"]
+
+
+class Perplexity:
```

Comment:
There seems to be a fair amount of repeated code between this class and Perplexity_torch. Can be done in a later PR but can we combine the two?